app/javascript/elm/tests/MobileSessionsFiltersTests.elm

```diff
@@ -15,6 +15,109 @@ import Test.Html.Selector as Slc
+        [ test "when ClosePopup is triggered the popup is hidden" <|
+            \_ ->
+                { defaultModel | popup = SelectFromItems [] }
+                    |> update ClosePopup
+                    |> Tuple.first
+                    |> view
+                    |> Query.fromHtml
+                    |> Query.hasNot [ Slc.id "popup" ]
+        , fuzz string "popup shows 4 main items" <|
+            \itemBase ->
+                let
+                    items =
+                        List.map (\index -> itemBase ++ String.fromInt index) (List.range 1 4)
+
+                    itemsHtml =
+                        items
+                            |> List.map
+                                (\item -> Slc.containing [ Slc.text item ])
+                in
+                items
+                    |> viewPopup SelectParameter False
+                    |> Query.fromHtml
+                    |> Query.has [ Slc.all itemsHtml ]
+        , fuzz string "popup shows only 4 main items when not extended" <|
+            \itemBase ->
+                let
+                    items =
+                        List.map (\index -> itemBase ++ String.fromInt index) (List.range 1 5)
+                in
+                items
+                    |> viewPopup SelectParameter False
+                    |> Query.fromHtml
+                    |> Query.findAll [ Slc.tag "li" ]
+                    |> Query.count (Expect.equal 4)
+        , test "popup has a button that triggers TogglePopupState" <|
+            \_ ->
+                []
+                    |> viewPopup SelectParameter False
+                    |> Query.fromHtml
+                    |> Query.find [ Slc.tag "button" ]
+                    |> Event.simulate Event.click
+                    |> Event.expect TogglePopupState
+        , test "TogglePopupState toggles the popup state" <|
+            \_ ->
+                defaultModel
+                    |> update TogglePopupState
+                    |> Tuple.first
+                    |> .isPopupExtended
+                    |> Expect.equal True
+        , fuzz string "popup shows all items when extended" <|
+            \itemBase ->
+                let
+                    items =
+                        List.map (\index -> itemBase ++ String.fromInt index) (List.range 1 5)
+                in
+                items
+                    |> viewPopup SelectParameter True
+                    |> Query.fromHtml
+                    |> Query.findAll [ Slc.tag "li" ]
+                    |> Query.count (Expect.equal 5)
+        , test "clicking on an item executed select function" <|
+            \_ ->
+                [ "item" ]
+                    |> viewPopup SelectParameter False
+                    |> Query.fromHtml
+                    |> Query.find [ Slc.tag "button", Slc.containing [ Slc.text "item" ] ]
+                    |> Event.simulate Event.click
+                    |> Event.expect (SelectParameter "item")
```

Comment:
Can I somehow create a fake select function? I had pass `SelectParameter ` cause `viewPopup` expects `(String -> Msg)`. But it should be able to accept any function of this type, so it's not cool that I had to use a specific one, right?

Reply:
I guess you could change the type to `(String -> msg)` in the function and pass from the test anything. Not sure though